apps/web/src/pages/Pools/AssessmentPlanBuilderPage/components/AssessmentDetailsDialog.tsx

```diff
@@ -355,13 +364,53 @@ const AssessmentDetailsDialog = ({
   const canAddScreeningQuestions =
     fields.length < SCREENING_QUESTIONS_MAX_QUESTIONS;
 
-  const assessedSkillsItems: CheckboxOption[] = alphaSortOptions(
-    allPoolSkills.map((poolSkill) => ({
-      value: poolSkill.id,
-      label: poolSkill?.skill?.name
-        ? getLocalizedName(poolSkill.skill.name, intl)
-        : intl.formatMessage(commonMessages.nameNotLoaded),
-    })),
+  const assessedSkillsItems: {
+    essentialSkillItems: CheckboxOption[];
+    assetSkills: CheckboxOption[];
+  } = allPoolSkills.reduce(
+    (
+      assessedSkills: {
+        essentialSkillItems: CheckboxOption[];
+        assetSkills: CheckboxOption[];
+      },
+      poolSkill,
+    ) => {
+      if (poolSkill.type === PoolSkillType.Essential) {
+        return {
+          essentialSkillItems: [
+            ...assessedSkills.essentialSkillItems,
+            {
+              value: poolSkill.id,
+              label: poolSkill?.skill?.name
+                ? getLocalizedName(poolSkill.skill.name, intl)
+                : intl.formatMessage(commonMessages.nameNotLoaded),
+            },
```

Comment:
Not necessary but, this is reused twice so we could move it to a helper like `poolSkillToOption` or something :woman_shrugging: I won't lose sleep over it though

Reply:
Good point 👍 Fixed by [5f94505](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9993/commits/5f945058bd6fd3b3f6d898a088ce4b9b973065bc)